sonar-plugin-api/src/main/java/org/sonar/api/batch/sensor/highlighting/internal/DefaultHighlighting.java

```diff
@@ -47,8 +48,7 @@ public DefaultHighlighting(@Nullable SensorStorage storage) {
     super(storage);
     syntaxHighlightingRuleSet = Sets.newTreeSet(new Ordering<SyntaxHighlightingRule>() {
       @Override
-      public int compare(@Nullable SyntaxHighlightingRule left,
-        @Nullable SyntaxHighlightingRule right) {
+      public int compare(@Nonnull SyntaxHighlightingRule left, @Nonnull SyntaxHighlightingRule right) {
```

Comment:
Is it not the workaround to override @Nullable annotation declared in interface Ordering?